src/reporter.cc

```diff
@@ -71,46 +78,109 @@ void MetadataReporter::ReportMetadata() {
 namespace {
 
 void SendMetadataRequest(std::vector<json::value>&& entries,
-                         const std::string& endpoint,
+                         const std::string& batch_uri,
+                         const std::string& publish_path,
                          const std::string& auth_header,
                          const std::string& user_agent,
                          bool verbose_logging)
     throw (boost::system::system_error) {
-  json::value update_metadata_request = json::object({
-    {"entries", json::array(std::move(entries))},
-  });
 
-  if (verbose_logging) {
-    LOG(INFO) << "About to send request: POST " << endpoint
-              << " User-Agent: " << user_agent
-              << " " << *update_metadata_request;
+  if (entries.size() == 1) {
+    // Add a copy of the single entry, except for the `views` field. This allows
+    // us to sent a request to the batch endpoint when we have a single request.
+    const json::Object* single_request = entries[0]->As<json::Object>();
+    json::value duplicate_request = json::object({  // ResourceMetadata
+      {"name", json::string(single_request->Get<json::String>("name"))},
```

Comment:
If you don't take my suggestion above (about erasing "views" from a copy of the request), at the very least these could be `single_request->At("name")->Clone()` (rather than accessing it as a string and then wrapping it back into a JSON string).

Reply:
Not `At` — [`std::map::at`](https://en.cppreference.com/w/cpp/container/map/at) (sorry for the typo). This would give you a reference to the actual `json::value` stored in the map.

Reply:
done